app/tournament/match.js

```diff
@@ -0,0 +1,39 @@
+export class Match {
+    constructor(round) {
+        this.round = round;
+        this.players = [];
+        this.isFinished = false;
+    }
+
+    addPlayer(player) {
+        if (this.players.length >= 2) {
+            throw new Error("Match has already 2 players");
+        }
+        this.players.push(player);
+    }
+
+    setNextMatch(match) {
+        this.nextMatch = match;
+    }
+
+    setWinner(player) {
```

Comment:
You still have to validate indexes, so validating the player belongs to the match is not harder.
And indexes are more error prone. If you pass the wrong player to a match, you detect the bug and throw. If you use indexes, you have less consistency.